.github/workflows/publish.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish
+on:
+  schedule:
+    - cron: "0 0 * * 0" # midnight UTC on Sunday
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```

Comment:
We'll have to figure out *who's* token this will be, and get it added by someone who has access to the repo settings. @nikomatsakis not sure if you do

Reply:
I have access, and I'm not averse to adding a token, but I do wonder if we can use a deploy key.

Reply:
Hmm, is there a reason why this is not using the builtin `${{ github.token }}`? Looking at the things you do in this job it seems to me that the builtin token has all the permissions required.

Reply:
Pushes from builtin token doesn't trigger workflows or webhooks. Is that okay with us?

Reply:
That's right, I forgot about that limitation.

I'd still prefer not to use custom tokens, as there is no way to limit which repos they have access unless we create a dedicate account for this. We have a base64-encoded deploy key in the `GITHUB_DEPLOY_KEY` secret, maybe you could use it to push the commits and tags?

Reply:
Can you describe a bit more about this `GITHUB_DEPLOY_KEY`? How is it created? I need to duplicate the process on my own repo so that I can test it.

Reply:
So, the contents of that variable is the base64-encoded copy of a private SSH key. You can generate a key with the same settings as the one configured in the repo with:

```
ssh-keygen -t ed25519 -a 100 -f key -q -N ""
```

Then you can get the contents of the variable with:

```
cat key | base64 -w 0
```

Finally you add the contents of `key.pub` in the "deploy keys" settings page of your repository.

Reply:
Github doesn't allow us to add a secret with `GITHUB_` prefix. Can you recheck and verify it is the name this repo has?

Secondly, is there any reason for it to be base64 encoded? I would need to decode it first but then I am unable to pass it to `checkout` action's `ssh-key` field since it will contain newlines.

You can check the attempt [here](https://github.com/pksunkara/chalk/runs/706720851?check_suite_focus=true)

Reply:
> Github doesn't allow us to add a secret with GITHUB_ prefix. Can you recheck and verify it is the name this repo has?

It is, the restriction was added after CI in this repository was configured.

> Secondly, is there any reason for it to be base64 encoded? I would need to decode it first but then I am unable to pass it to checkout action's ssh-key field since it will contain newlines.

The key was base64-encoded because the tooling in the rust-lang/simpleinfra repo expects it that way. I created another key in the `PUBLISH_DEPLOY_KEY` variable, which is not base64-encoded.

By the way, setting an environment variable with `::set-env::` makes its content public, so your workflow would've leaked the key if it worked.

Reply:
I can confirm that worked. Updated the PR.

> By the way, setting an environment variable with ::set-env:: makes its content public, so your workflow would've leaked the key if it worked.

Yeah, which was why I asking for the non base64 key. 😄